### ISSUE TITLE
Character heart meters

### ISSUE THREAD

**Issue by ExcaliburZero** [Infinite-Dango-Studios/ganbatte#7]
With dbe2d2484ed074536e3c3f235250118b5d2c8aac I added character heart meters to the game. Each character that is romance-able will have a heart meter that indicates their closeness to Haruki. (Except for Kosuke, [because](http://i3.kym-cdn.com/photos/images/newsfeed/000/680/203/ac5.jpg)) As the character becomes closer to Haruki the color of their heart meter will change.

For the heart meters we will need to decide on the colors that will be used and what order they will be in. And we will need to get together the artwork for them.

For the colors of the heart meter, we can use any set of colors we want to indicate the heart points. Though one example we can go by is the Harvest Moon series, which uses this order: Black < Purple < Blue < Green < Yellow < Orange < Red. We can go with that color system, alter it, or make a different one. However, I'm not sure how many levels there would need to be, and thus how many colors we would need.

We will then also need artwork for the character hearts. I'm not sure what you all want to do for that. If you just want to have a set design and do re-colors of it, have the heart get bigger for higher heart levels, have more intricate designs for higher levels, etc. We will need to come to decisions on those things and get the artwork ready.

This issue isn't very important currently, and will not need to be finished for the Proof of Concept.

We need to:
- [x] Decide upon the heart color order
- [x] Come to a decision on how the artwork for the different heart levels will work
- [ ] Get the artwork for the heart meters ready

**Comment by ExcaliburZero** [comment]
Also, here is a screenshot of the current implementation of heart meters. I will change the position and appearance of the heart meters as needed.

![sc - 02](https://cloud.githubusercontent.com/assets/4276196/6204499/75ff360c-b51b-11e4-9a59-7ef22946068a.png)

**Comment by Sabre-Hecate** [comment]
Black is ugly for a starting color. How about Blue? Then Yellow, Orange, Violet, Pink, Red, White. This is based upon Victorian Flower language, mostly. Yellow represents friendship. Orange represents passion or enthusiasm. Violet is escaping me at the moment but I think it's pure love. Pink is sweet and beautiful love. Red is deep, pure and passionate love. White roses symbolize marriage. I have NO idea how many levels we need. It might end up being: Blue -> Pink -> Purple -> Red -> White. Literally no idea. But white will be the highest level. And it'd be cool if it gets more elaborate as the level increases.

**Comment by ExcaliburZero** [comment]
@Sabre-Hecate 
Yeah, that sounds good. I think that color order should work well. We can decide how many colors to use later on, as it will probzbly be easier to figure out how many we need later on.

I think going with more elaborate designs for higher levels would likely make it a bit more interesting.

Also, note that the meters do not have to be hearts, they could be flowers or anything else like that.

**Comment by juanitaylor** [comment]
you guys figure out what you want and we artists will get on it

**Comment by Sabre-Hecate** [comment]
OK! Right now it looks like hearts, becoming more intricate. Or maybe Japanese cakes? Because Cafe? IDK

**Comment by ExcaliburZero** [comment]
I think hearts sound like a good idea. I'm not sure if Japanese cakes would be very apparent in their meaning.

**Comment by Sabre-Hecate** [comment]
Cakes was one of those things that sounded super cool for the 5 seconds after I posted it

**Comment by juanitaylor** [comment]
These I think I can do digitally. Exactly how many do we need and what colors?

**Comment by Sabre-Hecate** [comment]
We aren't certain on number >.< 

But color progression was decided to be:  Blue, Yellow, Orange, Violet, Pink, Red, White Should we even need that many. Maybe it starts at Orange..?

**Comment by juanitaylor** [comment]
I like yellow as a color too much to omit it XD

**Comment by Sabre-Hecate** [comment]
Well then :P

**Comment by juanitaylor** [comment]
• HYACINTH  Blue – Constancy

• ROSE  Yellow - Joy, Friendship

• TULIP Yellow - There's Sunshine in Your Smile
• POPPY  Red – Pleasure

• ORCHID  Cattleya - Mature Charm

• CAMELLIA  Pink - Longing for You

• ROSEBUD  Red - Pure and Lovely

• LILY  White - Virginity; Purity; Majesty; It's Heavenly to Be with You

I couldn't really find a good one for orange, so I'm doing yellow and red. Do these all look good to everyone?

**Comment by juanitaylor** [comment]
I've designed it so that they're hearts with flowers at the bottom, a-ok?
like this

![hni_0001](https://cloud.githubusercontent.com/assets/10875310/6319441/72d02ee0-ba8a-11e4-85a6-a3e9c3d7c9cc.JPG)

note that this is not the final design for the blue heart; I'll add blue hyacinths in place of the blue roses.

**Comment by juanitaylor** [comment]
Aslo, @ExcaliburZero , can we make them beat? I think that would be really cool, if they pulsed slightly~

**Comment by ExcaliburZero** [comment]
@juanitaylor 
We should be able make them beat as long as we have all the frames for the animation.

Also. the final images will need to be done on a transparent background.

**Comment by juanitaylor** [comment]
@ExcaliburZero 
so would I need to make separate versions of the hearts for animation frames?
eeep I don't know if my software can make transparent backgrounds???? that may be up to Priscilla, who is user @boxesofflowers on here, if you could please add her. She's agreed to help with the art, so maybe if I send it to her she can erase the bg or something

**Comment by ExcaliburZero** [comment]
@juanitaylor 
Yep, you would need to make all the different frames of the animation.

Any image that will be displayed on top of other images, like character art, icon, anything that isn't a background needs to have a transparent background, otherwise it won't work. The program you use may be able to do transparent backgrounds, but if not then you will likely need to use a different program.

I have added boxesofflowers to the list of collaborators.

**Comment by Sabre-Hecate** [comment]
Good work here with the hearts and flowers! Keep it up~! :dango: :dango:

**Comment by boxesofflowers** [comment]
@juanitaylor @ExcaliburZero I'm pretty familiar with animation, I'd just need to know how many fps, and I think Jacoba's heart is super duper but also in the sample, the heart looks pretty small, and I'd hate for the detail to go to waste. I can upload a file with colors if you'd like, I know Jacoba basically nailed them down (name wise) but I think it'd be nice if there was a physical file. :)

**Comment by ExcaliburZero** [comment]
@boxesofflowers 
I've started looking into implementing animations with Ren'Py, and it looks like Ren'Py does not support straight up animated gifs. However it does support basic animation in terms of switching between images. So any animation we use has to be simplistic, and more of a sporadic change than an animation. Like a heart beating by every ~.2 seconds switching between a slightly big and slightly small image.

Getting a set file for the colors sounds like a good idea.

**Comment by boxesofflowers** [comment]
![heart colors](https://cloud.githubusercontent.com/assets/11141654/6340160/c78af360-bb8a-11e4-8c9d-4911d9d53133.png)

ba-bang, color palette for those hearts ;)

**Comment by juanitaylor** [comment]
I
you guys figure it out I'll just leave these here
![hni_0003](https://cloud.githubusercontent.com/assets/10875310/6342404/7d8c4dec-bba6-11e4-95f1-19c92a46ec28.JPG)
![hni_0004](https://cloud.githubusercontent.com/assets/10875310/6342405/7d9b59d6-bba6-11e4-8dad-b2a45bd20ae5.JPG)
![hni_0005](https://cloud.githubusercontent.com/assets/10875310/6342406/7d9bfcb0-bba6-11e4-9997-6f314c26bf4e.JPG)
![hni_0006](https://cloud.githubusercontent.com/assets/10875310/6342408/7d9f6d50-bba6-11e4-960f-1ddba6302862.JPG)

Changed me concept a bit?

**Comment by juanitaylor** [comment]
The symbols in the middle are japanese numbers: ichi, ni, san, and shi/yon

**Comment by juanitaylor** [comment]
tell me I'm not doing great and I will punch you irl

**Comment by Sabre-Hecate** [comment]
Oooh, yes! I love these designs. I especially love the Japanese numbers!

**Comment by ExcaliburZero** [comment]
The designs look quite good so far.

**Comment by juanitaylor** [comment]
glory
thank you very much
I have all but the lillies done at this point
Stephen and I were talking about having buds that open as you progress??? I could work on the animation scenes for that, if other people were game.

**Comment by ExcaliburZero** [comment]
@juanitaylor 
What specifically do you mean by buds that open as you progress?

**Comment by juanitaylor** [comment]
like, I could illustrate closed flowers that open more as you get closer to the characters

**Comment by ExcaliburZero** [comment]
@juanitaylor 
Do you mean like the character hearts that are displayed when the character talks, or something else?

**Comment by juanitaylor** [comment]
yes, that's it
When you "level up" or whatever a special noise could sound or smth and the next layer of flowers would bloom around the heart

**Comment by ExcaliburZero** [comment]
@juanitaylor 
Okay, that makes sense.

I did some more research into animations with Ren'Py, and while Ren'Py is unable to run animated gif files, it is able to play videos and do basic coded animations, like a character blinking every few seconds.

**Comment by juanitaylor** [comment]
do you think we can make that work? if not, I won't bother with the closed flowers

**Comment by ExcaliburZero** [comment]
@juanitaylor 
What specifically is the animation that you want to make?

**Comment by juanitaylor** [comment]
the flowers opening

**Comment by ExcaliburZero** [comment]
@juanitaylor 
What in specifics do you want the animation to entail? Do you want the petals to slowly open? Do you want the petals to quickly open? Do you want a long animation or a short animation?

**Comment by ghost** [comment]
Hi, guys! So Jacoba wanted to know something about the meters; do we want buds on them, or no?

**Comment by ExcaliburZero** [comment]
@ChocoTacoKitty 
I'm not sure. What does she mean specifically by having buds on them? I know she was talking earlier about having animations where when a character's relationship level goes up it shows an animation with the heart where it starts with buds that then bloom. Is that what she meant?

**Comment by ghost** [comment]
@ExcaliburZero 
I believe that may be it. I can clarify with her real quick.

**Comment by ghost** [comment]
@ExcaliburZero 
Yes, that's what she meant.

**Comment by ExcaliburZero** [comment]
@ChocoTacoKitty 
In terms of animating it I'm not sure what she wants to do with that. I think it might work to have two frames (one with buds, one with flowers) and just use effects to transition between them.

**Comment by ghost** [comment]
@ExcaliburZero 
Ah, that makes sense. It's definitely a great idea, but I think that your idea would work best with it. I'll tell her.

**Comment by ghost** [comment]
@ExcaliburZero 
Okay, so I just told her about the idea and she told me that she could do three for better fluidity. Does that sound good to you?

**Comment by ExcaliburZero** [comment]
@ChocoTacoKitty 
Yep, three frames sounds good to me.

**Comment by ghost** [comment]
@ExcaliburZero 
Alright!

**Comment by juanitaylor** [comment]
HI guys, sorry about having Annissa as my intermediary, but I had no opportunity to get online, so I texted her. Thanks, @ChocoTacoKitty  for doing that for me~

So what I was thinking of doing is drawing 3 frames for each color heart- one with closed flowers, one with half-open flowers, and one with open flowers. I have all of the open flower ones finished, and I can post them??? sometime sooner rather than later

One thing though; do you think the blooming detail would detract from the game or go unnoticed? I don't want to do the extra work if it's not going to contribute positively.

**Comment by ExcaliburZero** [comment]
@juanitaylor 
That sounds good. Three frames should work.

I think the blooming animation should work and be noticeable if we add some effect to it to bring the player's attention to it. We could have sparkles, petals, etc. emanating from the heart as it blooms to draw the player's eye.

**Comment by juanitaylor** [comment]
That sounds like a good idea! Alright, I'll get on it.

**Comment by juanitaylor** [comment]
![hni_0010](https://cloud.githubusercontent.com/assets/10875310/6424481/004844be-bec9-11e4-8f54-3e2c92c06078.JPG)
![hni_0008](https://cloud.githubusercontent.com/assets/10875310/6424482/00545218-bec9-11e4-814a-6ee5c08e1a00.JPG)

EGGGGGGGGGGGGG
and that's the final white one btw

**Comment by Sabre-Hecate** [comment]
THE LAST ONE IS SO BEAUTIFUL

**Comment by juanitaylor** [comment]
which? egg or the flower heart?
I think both are true works of art

**Comment by ExcaliburZero** [comment]
So what should we do in terms of the heart meters?

I think we've decided on the colors, which is good. But we will need to wait till later on to see how many different colors we will actually use.

juanitayor has come up with some designs for the heart meters, however the sprites in the game will need have a transparent background. So I'm not sure if anyone wants to re-draw the hearts with a transparent background, or if we want to try to modify the existing files to have a transparent background. I'd recommend the former, as I tried making the backgrounds of the current images transparent, but it didn't come out very well.
